ecms_base/modules/custom/ecms_workflow/src/EcmsWorkflowBundleCreate.php

```diff
@@ -198,4 +223,64 @@ public function assignWorkflowToActiveTypes(): void {
 
   }
 
+  /**
+   * Add taxonomy permissions to the correct roles.
+   *
+   * @param string $taxonomyType
+   *   The machine name of the new taxonomy bundle.
+   */
+  public function addTaxonomyTypePermissions(string $taxonomyType): void {
+    $this->setTaxonomyRolePermissions($taxonomyType);
+  }
+
+  /**
+   * Set the correct role permissions for the taxonomy bundle.
+   *
+   * @param string $taxonomyType
+   *   The machine name of the new taxonomy bundle.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  private function setTaxonomyRolePermissions(string $taxonomyType): void {
+    $storage = $this->entityTypeManager->getStorage('user_role');
+
+    $site_admin_role = $storage->load(self::SITE_ADMIN_ROLE);
+
+    if (empty($site_admin_role)) {
+      return;
+    }
```

Comment:
If the site_admin role is empty for some reason, I think the publisher code should still have a chance to run. Can the site admin code be moved into this conditional, and change the logic to not empty?

Reply:
Reworked to continue processing if the role doesn't exist